File: ios/Classes/FlutterWebViewFactory.swift
import Foundation
import Flutter
import UIKit

public class FlutterWebViewFactory: NSObject, FlutterPlatformViewFactory {
    private var registrar: FlutterPluginRegistrar

    init(registrar: FlutterPluginRegistrar) {
        self.registrar = registrar
        super.init()
    }

    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }

    public func create(withFrame frame: CGRect, viewIdentifier viewId: Int64, arguments args: Any?) -> FlutterPlatformView {
        let arguments = args as? NSDictionary ?? [:]
        let channel = FlutterMethodChannel(name: "com.hisaichi5518/native_webview_\(viewId)", binaryMessenger: registrar.messenger())

        return FlutterWebViewController(
            parent: UIView(frame: frame),
            channel: channel,
            arguments: arguments,
            registrar: registrar
        )
    }
}

File: android/src/main/kotlin/com/hisaichi5518/native_webview/FlutterWebViewFactory.kt
package com.hisaichi5518.native_webview

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class FlutterWebViewFactory(
    private val messenger: BinaryMessenger
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val channel = MethodChannel(messenger, "com.hisaichi5518/native_webview_$viewId")

        return FlutterWebViewController(
            context,
            channel,
            args as Map<String, Any>
        )
    }
}